src/libgit2/checkout.c

```diff
@@ -516,9 +520,13 @@ static int checkout_action_with_wd(
 			*action = CHECKOUT_ACTION_IF(FORCE, UPDATE_BLOB, CONFLICT);
 		break;
 	case GIT_DELTA_DELETED: /* case 9 or 10 (or 26 but not really) */
-		if (checkout_is_workdir_modified(data, &delta->old_file, &delta->new_file, wd))
-			*action = CHECKOUT_ACTION_IF(FORCE, REMOVE, CONFLICT);
-		else
+		if (checkout_is_workdir_modified(data, &delta->old_file, &delta->new_file, wd)) {
+			if (git_iterator_current_skip_checkout(workdir)) {
```

Comment:
It's possible this logic doesn't come into play - if you have modified changes to a workdir file that shouldn't even be there, maybe `CONFLICT` is the right thing.

Reply:
That appears to be the case. I've removed it.